pandas/core/dtypes/dtypes.py

```diff
@@ -767,6 +767,7 @@ class PeriodDtype(ExtensionDtype, PandasExtensionDtype):
     _metadata = ('freq',)
     _match = re.compile(r"(P|p)eriod\[(?P<freq>.+)\]")
     _cache = {}  # type: Dict[str_type, PandasExtensionDtype]
+    freq = None
```

Comment:
Hmm this doesn't seem right - what error was this throwing before? Seems like this should shadow the property which wouldn't be desirable

Reply:
I added this pursuant to this thread: https://github.com/pandas-dev/pandas/pull/26071#discussion_r275087785

Reply:
Yea figured that was the case. Do you see any other resolutions? Having this as a class method and a property seems counterintuitive and the fact that it doesn't raise seems like a bug with mypy if anything

Reply:
Simply deleting the `freq = None` line does not create a new error, so that's a solution. Not sure why it threw errors before but not now.